data/spells/scripts/attack/envenom.lua

```diff
@@ -3,10 +3,69 @@ combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_EARTHDAMAGE)
 combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_CARNIPHILA)
 combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
 
-local condition = Condition(CONDITION_POISON)
-condition:setParameter(CONDITION_PARAM_DELAYED, true)
-condition:addDamage(25, 4000, -45)
-combat:setCondition(condition)
+local roundsTable = {
+	[20] = {1, 1},
+	[19] = {1, 2},
+	[18] = {1, 2},
+	[17] = {1, 2},
+	[16] = {1, 2},
+	[15] = {1, 2},
+	[14] = {1, 2},
+	[13] = {1, 2},
+	[12] = {1, 2},
+	[11] = {1, 2},
+	[10] = {1, 2},
+	[9] = {2, 3},
+	[8] = {2, 3},
+	[7] = {2, 3},
+	[6] = {3, 4},
+	[5] = {3, 4},
+	[4] = {4, 5},
+	[3] = {6, 7},
+	[2] = {10, 10},
+	[1] = {19, 20}
```

Comment:
Writing a function wouldn't prevent one to modify the output. You can easily modify the function to just `return 7` if you want. I think it gets cleaner, especially if other functions use the same table.